### ISSUE TITLE
Request to expose text track cues

### ISSUE THREAD

**Issue by forbesjo** [google/shaka-player#796]
This is a enhancement request for exposing a text track cues. The purpose of this request is to allow for custom text track display.

**Comment by joeyparrish** [comment]
Makes sense to me.  I'll tentatively schedule it for v2.2, but it is lower priority than bug fixes and HLS work.  We will try to come up with a design first, publish a design doc, then implement.

**Comment by joeyparrish** [comment]
In #886, @hghazzi asks to know which Player instance cues came from.  I've marked that as a duplicate and I'm adding a requirement here that there should be some way to know which Player instance is associated with the display instance.

If the text display factory is configurable per-player (as opposed to some global registration), this requirement could be met by the application itself.  This is a very lazy example of what I mean, and does not reflect the final interface being designed by @ismena:

```js
player.configure({
  textDisplayFactory: function() {
    return new MyCustomTextDisplay(player);
  }
});
```

Then, any call the Player makes to `new config.textDisplayFactory()` will create an instance of `MyCustomTextDisplay` which knows exactly which `Player` instance it belongs to.

**Comment by hghazzi** [comment]
@joeyparrish We would like to collaborate with your team to develop the new implementation of the text tracks plugin. 

Not sure when the timeline for 2.2.0 is, but we would really like to help out and get it in there since our current workaround does not work well when we have multiple players on a page.

**Comment by ismena** [comment]
@hghazzi Sorry for the delay, I'm working on this as I write this (l literally have the CL on my other monitor this second :)).
I'll try to get it into code review this week or early next week and bully the rest of the team to review it ASAP :)

**Comment by hghazzi** [comment]
Awesome! Can you add me as a reviewer or let me know when a PR is up? Would love to take a look.

**Comment by ismena** [comment]
Unfortunately we can't add external reviewers to the team's code reviews, but I'll make sure to let you know as soon as it's out.

**Comment by ismena** [comment]
@hghazzi Just checked this in! We'll be writing an upgrade guide with more information on the new parsing/displaying logic, but please feel free to take a look at the commit and let me know if you have any question in the meantime.

Basically it's what Joey said it would be in the comment above :)

**Comment by hghazzi** [comment]
@ismena So this change is in master now? Is there a timeline for v2.2.0?

**Comment by ismena** [comment]
Yes! We also have an upgrade guide which (hopefully) explains how to use it: https://github.com/google/shaka-player/blob/master/docs/tutorials/upgrade-v2-1.md

For v2.2 I was about to say we only had one issue left, so probably early next week, but then I saw that @joeyparrish scheduled a few more, so I'm going to point a finger at him to give the new timeline :P

**Comment by hghazzi** [comment]
@joeyparrish Do you have an updated timeline for `v2.2`? And thanks for the upgrade guide, looks like the parser updates will give us what we need to fix our issues!

**Comment by joeyparrish** [comment]
@hghazzi, we will be releasing v2.2.0 as soon as #960 is resolved.  I estimate that this may take another week or two, but it is difficult to say for certain.  Sorry for the long delays.

**Comment by joeyparrish** [comment]
@hghazzi, we have now closed all the issues we had planned for v2.2.0.  Once we take some extra time for QA, we will release v2.2.0.  Thanks!
